File: src/main/julia/day01.jl
using Printf

masses = map(x->parse(Int, x), readlines("input01.txt"))

fuel(x) = floor(x / 3 - 2)
fuel(1969)

function fuel2(x) 
    f = fuel(x)
    f < 0 ? 0 : f + fuel2(f)
end    

fuel2(1969)

@printf "part 1: %10.0f\n" sum(map(x->fuel(x), masses))
@printf "part 2: %10.0f\n" sum(map(x->fuel2(x), masses))
# exit()

File: src/main/python/2019/day01.py

with open("input01.txt") as f:
    data = f.readlines()
    data = [int(x.strip()) for x in data] 


def fuel(m):
    return int((m/3)-2)

def fuel2(m):
    f = fuel(m)
    return f + fuel2(f) if f>0 else 0

print(sum(map(fuel, data), 0))

print(sum(map(fuel2, data), 0))
